src/main/kotlin/com/openlattice/mechanic/Mechanic.kt

```diff
@@ -194,6 +204,19 @@ class Mechanic {
         }
     }
 
+    fun regenerate(regenNames: Set<String>) {
+        val regens = context.getBeansOfType(Regeneration::class.java)
```

Comment:
Can replace this and `doUpgrade` with a generic method that takes a class and some logging templates

Reply:
Unless we specify a common interface for them, I wouldn't do this in a generic method

Reply:
👍 no worries, this is fine for now